src/lib/holocene/markdown-editor/markdown-preview.svelte

```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
+  import { type ClassNameValue, twMerge } from 'tailwind-merge';
+
+  import { page } from '$app/state';
+
+  import { useDarkMode } from '$lib/utilities/dark-mode';
+
+  interface Props {
+    content: string;
+    class?: ClassNameValue;
+    overrideTheme?:
+      | 'background'
+      | 'primary'
+      | 'info'
+      | 'details'
+      | ''
+      | undefined;
+  }
+
+  let { content, class: className = '', overrideTheme = '' }: Props = $props();
+
+  let iframe: HTMLIFrameElement | null = $state(null);
+
+  const resizeIframe = () => {
+    if (!iframe) return;
+    const minHeight = 100;
+    const height = Math.max(
+      iframe.contentWindow.document.body.scrollHeight + 2,
+      minHeight,
+    );
+    iframe.height = '';
+    iframe.height = height + 'px';
+  };
+
+  const { workflow: workflowId, run: runId, namespace } = page.params;
+
+  const replaceTemplate = (content: string) => {
+    if (namespace) {
+      content = content.replace(/\{namespace\}/g, namespace);
+    }
+    if (workflowId) {
+      content = content.replace(/\{workflowId\}/g, workflowId);
+    }
+    if (runId) {
+      content = content.replace(/\{runId\}/g, runId);
+    }
+    return content;
+  };
+
+  const templatedContent = $derived(replaceTemplate(content));
+  const previewTheme = $derived($useDarkMode ? 'dark' : 'light');
+  const previewPath = $derived(
+    `/render?content=${encodeURIComponent(templatedContent)}&theme=${previewTheme}&overrideTheme=${overrideTheme}`,
+  );
+</script>
+
+<section class={twMerge('h-full w-full', className)}>
+  <iframe
+    bind:this={iframe}
+    onload={resizeIframe}
```

Comment:
Do we care if the window size is changed  and need to update this? Should we use a resizeObserver (or whatever the window version is) on this to keep things properly laid out?

Reply:
Good question, this was existing code I copied over from [src/lib/holocene/monaco/markdown.svelte](https://github.com/temporalio/ui/pull/2920/files#diff-224b769364bab6f5d9cc0529ec36ffb2d0c32a7e38bb7212e39759cb719c6276), so I didn't want to touch it.

Reply:
It seems to work fine

https://github.com/user-attachments/assets/cd4166f8-8d46-4e55-aa7a-d557a681cf4f